contrib/files-touched-check

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+set -e -o pipefail
+export TZ=UTC
+
+while read -r status name; do
+
+    case "$name" in
+        README.md) continue ;;
+        contrib/*) continue ;;
+        21.99-guixtest1/*) continue ;;  # guixtest1 used a different hierarchy
+    esac
+    
+    if [ "$status" != "A" ]; then
+        echo "ERR: File status is not 'A' (for add): '$status'"
+        exit 1
+    fi
+
+    if [[ $name =~ ^[^/]+/[^/]+/[^/]+.SHA256SUMS(|.asc)$ ]]; then
+        if [ -z "$last_prefix" ]; then
+            last_prefix="$(dirname "$name")"
+            continue
+        elif [ "$last_prefix" = "$(dirname "$name")" ]; then
+            continue
+        else
+            echo "ERR: Added files need to be under the same prefix: '${last_prefix}'"
+            exit 1
+        fi
+    else
+        echo "ERR: Added unknown file '$name'"
+        exit 1
+    fi
+
+done < <(git diff --no-commit-id --name-status -r "${1:-${CIRRUS_BASE_SHA}...HEAD}")
```

Comment:
Would be nice to write this in python, because bash is so unreliable.

Currently this fails, but returns normally (green).


```
./contrib/files-touched-check
./contrib/files-touched-check: line 34: git: command not found